src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  active: boolean;
+  selectedUserId: number | null;
+  onClose: () => void;
+  selectedTodo: Todo | null;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  active,
+  selectedUserId,
+  onClose,
+  selectedTodo,
+}) => {
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (selectedUserId !== null) {
+      setLoading(true);
+      getUser(selectedUserId)
+        .then(data => setUser(data))
+        .finally(() => setLoading(false));
```

Comment:
The `getUser` API call lacks error handling. Consider adding a `.catch` block to handle any potential errors during the data fetching process. This will ensure the application can gracefully manage any issues that arise.